### PR TITLE
Make transaction name conflict check more robust

### PR DESCRIPTION
**Pull request by anand1976**
The `PessimisticTransaction::SetName()` code checks for an existing txn of the given name before registering the new txn. However, this is not atomic, which could result in a race condition if two txns try to register with the same name. Both might succeed and lead to unpredictable behavior. This PR makes the test and set atomic.

### CODE REVIEW

**Review by jowlyzhang — APPROVED** [review]
LGTM! Thanks for the quick fix. There is some formatting needed.

File: utilities/transactions/pessimistic_transaction_db.cc [hunk 1]
```diff
@@ -755,13 +760,17 @@ void PessimisticTransactionDB::SetDeadlockInfoBufferSize(uint32_t target_size) {
...
 }
 
-void PessimisticTransactionDB::RegisterTransaction(Transaction* txn) {
+Status PessimisticTransactionDB::RegisterTransaction(Transaction* txn) {
   assert(txn);
   assert(txn->GetName().length() > 0);
-  assert(GetTransactionByName(txn->GetName()) == nullptr);
   assert(txn->GetState() == Transaction::STARTED);
   std::lock_guard<std::mutex> lock(name_map_mutex_);
+  if (GetTransactionByNameLocked(txn->GetName()) != nullptr) {
```
Comment (pdillinger) [inline utilities/transactions/pessimistic_transaction_db.cc:768]: This is an extra map lookup that can be avoided, using `insert`.

**Review by pdillinger — CHANGES_REQUESTED** [review]

File: utilities/transactions/pessimistic_transaction_db.cc [hunk 2]
```diff
@@ -723,6 +723,11 @@ void PessimisticTransactionDB::ReinitializeTransaction(
 Transaction* PessimisticTransactionDB::GetTransactionByName(
     const TransactionName& name) {
   std::lock_guard<std::mutex> lock(name_map_mutex_);
+  return GetTransactionByNameLocked(name);
+}
+
+Transaction* PessimisticTransactionDB::GetTransactionByNameLocked(
```
Comment (pdillinger) [inline utilities/transactions/pessimistic_transaction_db.cc:729]: Not strictly required, but totally OK

**Review by pdillinger — APPROVED** [review]
Awesome, thanks!

I see an existing unit test of duplicate name in TransactionTest.TwoPhaseNameTest 👍

**Pull request closed without merging**